src/disco/containers/InstallButton.js

```diff
@@ -29,12 +32,53 @@ export class InstallButton extends React.Component {
     downloadProgress: 0,
   }
 
+  componentDidMount() {
+    const { dispatch, guid, installUrl, slug } = this.props;
+    this.addonManager = new AddonManager(guid, installUrl, this.statusChanged);
+    this.addonManager.getAddon().then(() => {
```

Comment:
Can you move the `then` on the next line - makes the flow clearer.

Also I reckon it's more obvious to use `catch()` instead of the reject arg.